### DISCUSSION TITLE
"captions" node in the configuration

### DISCUSSION THREAD

**Discussion by stockiNail** [kurkle/chartjs-chart-treemap#53] [Ideas]
As done by PR #51, I think it could be helpful to configure also the captions.

This is how the config will look like:

```javascript
  captions: {
    align: undefined,
    color: undefined,
    display: false,
    formatter: (ctx) => ctx.raw.g,
    font: {},
    padding: 3
  },
```

The options `groupLabels` should be removed in favor to `captions.display` .
I have already implemented in my fork and I'm testing it.
Let me know if makes sense.

**Comment by stockiNail** [comment]
In my implementation, I have removed the `formatter` and the label can show only the group name.

```javascript
TreemapElement.defaults = {
  borderSkipped: undefined,
  borderWidth: undefined,
  dividerCapStyle: 'butt',
  dividerColor: 'black',
  dividerDash: undefined,
  dividerDashOffset: 0,
  dividerWidth: 0,
  groupDividers: false,
  spacing: undefined,
  label: undefined,
  rtl: undefined,
  captions: {
    align: undefined,
    color: undefined,
    display: true,
    font: {},
    padding: 3
  },
  labels: {
    align: 'center',
    color: undefined,
    display: false,
    formatter: (ctx) => ctx.raw.g ? [ctx.raw.g, ctx.raw.v] : ctx.raw.v,
    font: {},
    position: 'middle',
    padding: 3
  }
};
```

**Comment by kurkle** [comment]
Looks good, I'd be happy to review a PR for that!

**Reply by stockiNail** [reply]
> 
> 
> Looks good, I'd be happy to review a PR for that!

Fine! Let me finish that PR #51 (I'm working on the documentation, hoping to commit it soon), because this PR is implemented basing on the other.

**Reply by stockiNail** [reply]
> 
> 
> Looks good, I'd be happy to review a PR for that!

PR #54 submitted. Doc is missing but coming soon (if PR is ok)
